### DISCUSSION TITLE
PyBaMM Models for Battery Fault Diagnosis or Prognosis

### DISCUSSION THREAD

**Discussion by Akila1993** [pybamm-team/PyBaMM#2770] [Q&A]
I am working on battery fault prognosis and control based on model based methods. Is there any similar work done using the PyBaMM battery model. Or Can someone suggest me any references please?

**Comment by brosaplanella** [comment]
I guess the best reference for degradation modelling in PyBaMM is the article by [O'Kane et al. (2022)](https://pubs.rsc.org/en/content/articlehtml/2022/cp/d2cp00417h).

**Reply by Mrzhang-hub** [reply]
It seems that no one has been using pybamm to do battery control such as lithium plating control during fast charging. I wanna know the challenges when applying these pybamm models,such as SPMe, into the battery control framework.

**Reply by Akila1993** [reply]
I'd also like to know if PyBaMM can be applied to real-time control aspects in BMS like fault detection or fault estimation.

**Reply by valentinsulzer** [reply]
You can use the PyBaMM model as a reference (model in the loop) instead of a real battery, but for real-world BMS control it's better to develop your own algorithm in C or Simulink, since the PyBaMM models aren't optimized to run on an embedded controller.

**Comment by Akila1993** [comment]
> It seems that no one has been using pybamm to do battery control such as lithium plating control during fast charging. I wanna know the challenges when applying these pybamm models,such as SPMe, into the battery control framework.
